code/modules/client/preference_setup/general/02_body.dm

```diff
@@ -109,6 +109,8 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 	pref.h_style		= sanitize_inlist(pref.h_style, GLOB.hair_styles_list, initial(pref.h_style))
 	pref.f_style		= sanitize_inlist(pref.f_style, GLOB.facial_hair_styles_list, initial(pref.f_style))
 	pref.b_type			= sanitize_text(pref.b_type, initial(pref.b_type))
+	if(!pref.b_type || !(pref.b_type in global.valid_bloodtypes))
+		pref.b_type = pick(global.valid_bloodtypes)
```

Comment:
There's actually a `RANDOM_BLOOD_TYPE` macro that weights it, I assume based on RL statistics